compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -1109,6 +1117,63 @@ public BLangNode transform(AnnotationNode annotation) {
         return bLAnnotationAttachment;
     }
 
+    @Override
+    public BLangNode transform(AnnotationDeclarationNode annotationDeclarationNode) {
+        BLangAnnotation annotationDecl = (BLangAnnotation) TreeBuilder.createAnnotationNode();
+        DiagnosticPos pos = getPosition(annotationDeclarationNode);
+        annotationDecl.pos = pos;
+        annotationDecl.name = createIdentifier(pos, annotationDeclarationNode.annotationTag().text());
+
+        if (annotationDeclarationNode.visibilityQualifier() != null) {
+            annotationDecl.addFlag(Flag.PUBLIC);
+        }
+
+        if (annotationDeclarationNode.constKeyword() != null) {
+            annotationDecl.addFlag(Flag.CONSTANT);
+        }
+
+        annotationDecl.annAttachments = applyAll(annotationDeclarationNode.metadata().annotations());
+
+        annotationDeclarationNode.metadata().documentationString().ifPresent(docString-> {
+            annotationDecl.markdownDocumentationAttachment = createMarkdownDocumentationAttachment(docString);
+        });
+
+        Node typedesc = annotationDeclarationNode.typeDescriptor();
+        if (typedesc != null) {
+            annotationDecl.typeNode = createTypeNode(typedesc);
+        }
+
+        SeparatedNodeList<Node> paramList = annotationDeclarationNode.attachPoints();
+
+        for (Node child : paramList) {
+            AnnotationAttachPointNode attachPoint = (AnnotationAttachPointNode) child;
+            boolean source = attachPoint.sourceKeyword() != null;
+            AttachPoint bLAttachPoint;
+            String firstIndent = attachPoint.firstIdent().text().toLowerCase();
+            if (firstIndent == "object") {
+                switch (attachPoint.secondIdent().text().toLowerCase()) {
+                    case "function" :
+                        bLAttachPoint = AttachPoint.getAttachmentPoint("objectfunction", source);
+                        break;
+                    case "field" :
+                        bLAttachPoint = AttachPoint.getAttachmentPoint("objectfield", source);
+                        break;
+                    default:
+                        bLAttachPoint = AttachPoint.getAttachmentPoint("objecttype", source);
+                }
+            } else if (firstIndent == "resource") {
+                bLAttachPoint = AttachPoint.getAttachmentPoint("resourcefunction", source);
+            } else if (firstIndent == "record") {
+                bLAttachPoint = AttachPoint.getAttachmentPoint("recordfield", source);
+            } else {
```

Comment:
Can use a switch case for the outer if-else as well.

Reply:
Done